src/rest/RequestHandler.js

```diff
@@ -177,7 +177,7 @@ class RequestHandler {
     }
 
     let sublimitTimeout;
-    if (res && res.headers) {
+    if (res?.headers) {
```

Comment:
Here's a genuine question, when will res not be defined 👀

Reply:
In #6758 maybe?

Reply:
It's not even related and we clearly get a res because the DiscordAPIError emitted has the HTTP status code

Reply:
I’m not sure where that status code is coming from though because cloudflare’s error is 1020 and you get no json object back at all so it sort of seems like it’s made up somewhere along the way

Reply:
I don't see how `res` could be anything other than a `Response` at this point, I'll remove the optional chain

Typings:
```ts
declare function fetch(
    url: RequestInfo,
    init?: RequestInit
): Promise<Response>;
```

Reply:
yeah, the next few if statements all use res. so if it was not defined we'd have gotten errors long ago